otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/ama-metrics-collector-hpa.yaml

```diff
@@ -1,4 +1,5 @@
-{{- if and (ne .Values.AzureMonitorMetrics.ArcExtension true) (.Values.AzureMonitorMetrics.CollectorHPAEnabled)}}
+{{- $arcExtensionSettings := include "arc-extension-settings" . | fromYaml }}
+{{- if $arcExtensionSettings.hpaEnabled }}
```

Comment:
grace, isn't the only way we are making the customer control replica count is thru `deploymentReplicas` param for chart ? So this merge will not evaluate to true for arc ?

Reply:
Right, exactly. So the logic for this in the helper template is

```
hpaEnabled: true
{{- if or $isArcExtension (ne .Values.AzureMonitorMetrics.CollectorHPAEnabled true) }}
    hpaEnabled: false
{{- end }}
```

Reply:
ok thanks. all these new params will be accessible thru arc cli, i am assuming (2 mount settings to enable/disable mounts and also deployment replicas). Also , the deployment replicas isn't synced to goal state by arc extension? is it also not validated? can they have it as 1 or say 21 ?